src/librustc_typeck/diagnostics.rs

```diff
@@ -211,6 +211,53 @@ Reference:
 http://doc.rust-lang.org/reference.html#trait-objects
 "##,
 
+E0036: r##"
+This error occurred when you pass too many or not enough type parameters to a
+method. Bad example:
+
+```
+struct Test;
+
+impl Test {
+    fn method<T>(&self, v: &[T]) -> usize {
+        v.len()
+    }
+}
+
+fn main() {
+    let x = Test;
+    let v = &[0i32];
+
+    x.method::<i32, i32>(v); // not good: only one type parameter is expected!
+}
+```
+
+To fix it, just remove/add type parameter to have the good number:
```

Comment:
Instead of removing or adding type parameters, I’d suggest to just “specify a correct number of type parameters”.